packages/block-library/src/post-comment-avatar/edit.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	InspectorControls,
+	__experimentalUseBorderProps as useBorderProps,
+	__experimentalImageSizeControl as ImageSizeControl,
+	useBlockProps,
+} from '@wordpress/block-editor';
+import { PanelBody, ResizableBox } from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import { __, isRTL } from '@wordpress/i18n';
+
+export default ( { attributes, context, setAttributes } ) => {
+	const { className, style, height, width } = attributes;
+	const { commentId } = context;
+
+	// Maximum provided by the WordPress site.
+	const maxImageWidth = 96;
+	const maxImageHeight = 96;
+	if ( ! height || ! width ) {
+		setAttributes( {
+			height: maxImageHeight,
+			width: maxImageWidth,
+		} );
+	}
+
+	const displayAvatar = useSelect( ( select ) => {
+		const { getEntityRecord } = select( coreStore );
+		const comment = getEntityRecord( 'root', 'comment', commentId );
+		const authorAvatarUrls = comment?.author_avatar_urls; // eslint-disable-line camelcase
+		const borderProps = useBorderProps( attributes );
```

Comment:
We shouldn't have the hook inside the the `useSelect` and neither return the component from here. This is not obvious now because we have not any dependency in `useSelect` so it always runs. But we do need a dependency in `useSelect`, the `[ commentId ]`.

Reply:
I updated the code, using the query-inspector-controls block as reference. Let me know please if it is OK!